google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClientImpl.java

```diff
@@ -301,17 +218,13 @@ public Long apply(Session session) {
     }
   }
 
-  private <T> T runWithSessionRetry(SessionMode mode, Function<Session, T> callable) {
-    PooledSessionFuture session =
-        mode == SessionMode.READ_WRITE ? getReadWriteSession() : getReadSession();
+  private <T> T runWithSessionRetry(Function<Session, T> callable) {
+    PooledSessionFuture session = getReadSession();
```

Comment:
nit: could we rename methods such as `getReadSession()` to `getSession()` now that we do not have such separation?

Reply:
Done.